sqld/src/replication/primary/frame_stream.rs

```diff
@@ -13,25 +13,27 @@ use crate::replication::{FrameNo, LogReadError, ReplicationLogger};
 pub struct FrameStream {
     pub(crate) current_frame_no: FrameNo,
     pub(crate) max_available_frame_no: FrameNo,
-    logger: Arc<ReplicationLogger>,
+    logger: Weak<ReplicationLogger>,
     state: FrameStreamState,
     wait_for_more: bool,
 }
 
 impl FrameStream {
     pub fn new(
-        logger: Arc<ReplicationLogger>,
+        logger: Weak<ReplicationLogger>,
```

Comment:
I don't really like using weak I think it opens up a lot of opportunity to run into issues (and Ive seen it happen plenty in the past). Can we write this in such a way without weak? Or atleast can we just do all of the arc stuff within the logger so we don't expose this everywhere?

Reply:
yeah, maybe this is a bit too brittle. Let me come up with something better